src/main/java/net/staticsnow/nexus/repository/apt/internal/AptRecipeSupport.java

```diff
@@ -0,0 +1,182 @@
+
+import static org.sonatype.nexus.repository.http.HttpMethods.DELETE;
+import static org.sonatype.nexus.repository.http.HttpMethods.GET;
+import static org.sonatype.nexus.repository.http.HttpMethods.HEAD;
+import static org.sonatype.nexus.repository.http.HttpMethods.MKCOL;
+import static org.sonatype.nexus.repository.http.HttpMethods.POST;
+import static org.sonatype.nexus.repository.http.HttpMethods.PUT;
+
+public abstract class AptRecipeSupport
+  extends RecipeSupport
+{
+  @Inject
+  protected Provider<AptSecurityFacet> securityFacet;
+
+  @Inject
+  protected Provider<ConfigurableViewFacet> viewFacet;
+
+  @Inject
+  protected Provider<HttpClientFacet> httpClientFacet;
+
+  @Inject
+  protected Provider<NegativeCacheFacet> negativeCacheFacet;
+
+  @Inject
+  protected Provider<AptFacetImpl> aptFacet;
+
+  @Inject
+  protected Provider<StorageFacet> storageFacet;
+
+  @Inject
+  protected Provider<AttributesFacet> attributesFacet;
+
+  @Inject
+  protected Provider<SingleAssetComponentMaintenance> componentMaintenance;
+
+  @Inject
+  protected Provider<SearchFacet> searchFacet;
+
+  @Inject
+  protected Provider<PurgeUnusedFacet> purgeUnusedFacet;
+
+  @Inject
+  protected ExceptionHandler exceptionHandler;
+
+  @Inject
+  protected TimingHandler timingHandler;
+
+  @Inject
+  protected SecurityHandler securityHandler;
+
+  @Inject
+  protected NegativeCacheHandler negativeCacheHandler;
+
+  @Inject
+  protected PartialFetchHandler partialFetchHandler;
+
+  @Inject
+  protected UnitOfWorkHandler unitOfWorkHandler;
+
+  @Inject
+  protected ConditionalRequestHandler conditionalRequestHandler;
+
+  @Inject
+  protected ContentHeadersHandler contentHeadersHandler;
+
+  @Inject
+  protected AptSnapshotHandler snapshotHandler;
+
+  protected AptRecipeSupport(final Type type, final Format format) {
+    super(type, format);
+  }
+
+  protected static Builder assetsMatcher() {
+    return new Builder().matcher(
+        LogicMatchers.and(
+            new ActionMatcher(GET, HEAD),
+            new TokenMatcher("/{path:.+}")
+        ));
+  }
+
+  protected static Builder rebuildIndexMatcher() {
+    return new Builder().matcher(
+        LogicMatchers.and(
+            new ActionMatcher(POST),
+            new TokenMatcher("/rebuild-indexes")
+        ));
+  }
+
+  protected static Builder otherMatcher() {
+    return new Builder().matcher(
+        LogicMatchers.and(
+            new ActionMatcher(POST),
+            new TokenMatcher("/")
+        ));
+  }
+
+  protected static Builder signingMatcher() {
+    return new Builder().matcher(
+        LogicMatchers.and(
+            new ActionMatcher(GET),
+            new TokenMatcher("/repository-key.gpg")
+        ));
+  }
+
+  protected static Builder snapshotMatcher() {
+    return new Builder().matcher(
+        LogicMatchers.and(
+            new ActionMatcher(GET, HEAD),
+            new TokenMatcher("/snapshots/{id:([^/]+)}/?{path:(.*)}")
```

Comment:
You'll want to ensure this RegEx works, I couldn't really test it per se. This should pass id and path as the groups to the handler if I've done it right, which should make for some cleaner code.